### PR TITLE
Another grab bag of compiler optimizations

### PR DESCRIPTION
**Pull request by retronym**
Review by @gkossakowski

### CODE REVIEW

**Comment by retronym** [comment]
PLS REBUILD/pr-scala@e3af86a

```
[info] [info] ! Prop.within: Falsified after 73 passed tests.
[info] [info] > Labels of failing property: 
[info] [info] Timeout
[info] [info] > ARG_0: 100
[info] [info] > ARG_1: 10
[info] [info] + Prop.sizedProp: OK, passed 100 tests.
[info] [info] + Prop.someFailing: OK, passed 100 tests.
```

**Comment by gkossakowski** [comment]
PLS REBUILD ALL

Somehow PR validator got stuck on this one.

**Comment by retronym** [comment]
Review by @gkossakowski, please

**Comment by gkossakowski** [comment]
LGTM.

File: src/compiler/scala/tools/nsc/transform/Erasure.scala [hunk 1]
```diff
@@ -92,11 +92,22 @@ abstract class Erasure extends AddInterfaces
   // more rigorous way up front rather than catching it after the fact,
   // but that will be more involved.
   private def dotCleanup(sig: String): String = {
+    // OPT 50% of time in generic signatures (~1% of compile time) was in this method, hence the imperative rewrite.
     var last: Char = '\u0000'
-    sig map {
-      case '.' if last != '>' => last = '.' ; '$'
-      case ch                 => last = ch ; ch
+    var i = 0
+    val len = sig.length
```
Comment (xuwei-k) [inline src/compiler/scala/tools/nsc/transform/Erasure.scala:98]: unused variable?

Comment (retronym) [inline src/compiler/scala/tools/nsc/transform/Erasure.scala:98]: good spot, I guess I meant to use it in the `while`.

**Pull request merged**